sql/wsrep_mysqld.cc

```diff
@@ -2242,6 +2295,232 @@ static void wsrep_TOI_end(THD *thd) {
   }
 }
 
+static int wsrep_NBO_begin_phase_one(THD *thd, const char *db_,
+                                     const char *table_,
+                                     const TABLE_LIST *table_list,
+                                     dd::Tablespace_table_ref_vec *trefs,
+                                     Alter_info *alter_info,
+                                     wsrep::key_array *fk_tables) {
+  uchar *buf(0);
+  size_t buf_len(0);
+  int buf_err;
+  int rc;
+
+  WSREP_DEBUG("Starting NBO phase one");
+
+  mysql_mutex_lock(&thd->LOCK_thd_data);
+
+  bool is_thd_killed = thd->is_killed();
+
+  mysql_mutex_unlock(&thd->LOCK_thd_data);
+
+  thd->wsrep_skip_wsrep_hton = true;
```

Comment:
In other places, `wsrep_skip_wsrep_hton` is set under the protection of `thd->LOCK_thd_data`.
See `wsrep_to_isolation_end()` in `sql/wsrep_mysqld.cc`
``` 
  2694   DEBUG_SYNC(thd, "wsrep_to_isolation_end_before_wsrep_skip_wsrep_hton");


  2695   mysql_mutex_lock(&thd->LOCK_thd_data);
  2696 
  2697   thd->wsrep_skip_wsrep_hton = false;
  2698 
  2699   mysql_mutex_unlock(&thd->LOCK_thd_data);
  2700   DEBUG_SYNC(thd, "wsrep_to_isolation_end_after_wsrep_skip_wsrep_hton");
```
```suggestion
  thd->wsrep_skip_wsrep_hton = true;
  mysql_mutex_unlock(&thd->LOCK_thd_data);
```

Reply:
Fixed.